pandas/_libs/lib.pyx

```diff
@@ -1171,6 +1172,9 @@ def infer_dtype(object value, bint skipna=False):
         values = construct_1d_object_array_from_listlike(value)
 
     values = getattr(values, 'values', values)
+    if skipna:
+        values = values[~isnaobj(values)]
```

Comment:
this is a python and not a cimport, why are you not using checknull?

Reply:
so this isn array, ok, then add ``isnaobj`` to ``missing.pxd`` and make it a ``cpdef``. then you can cimport it. (and you need to type return value)